src/middlewares/verifyToken.ts

```diff
@@ -35,11 +36,17 @@ const verifyToken = (tokenType: string) => {
       }
 
       // Decode the token to get the payload
-      const decodedToken = jwt.verify(token, SECRET_KEY) as CustomJwtPayload;
-      if (!decodedToken) throw new AppError('Invalid token', StatusCodes.UNAUTHORIZED);
+      jwt.verify(
+        token,
+        SECRET_KEY,
+        (err: JwtVerifyCallbackError, decoded: JwtVerifyCallbackDecoded) => {
+          if (err) throw new AppError('Invalid token', StatusCodes.UNAUTHORIZED);
 
-      // Attach the decoded user information to the request object
-      req.user = { userId: decodedToken.id, email: decodedToken.email };
+          // Attach the decoded user information to the object
+          const decodedToken = decoded as CustomJwtPayload;
```

Comment:
@RiteshAdwani Is there a way to avoid this type assertion?

Reply:
@adarshpandya-simform Thanks for pointing out, actually the "decoded" comes from an untrusted source, so we can't actually give a type to the decoded value. We can either use type assertion or we can make use of Type predicate. The latest commit avoids type assertion and instead makes use of type predicate.